File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/context/fight/GameFightNewRoundMessage.scala





















// Generated on 05/03/2013 10:34:40
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait GameFightNewRoundMessage extends MessageDeserializer {
  val messageId = 6239

  protected trait Message extends BaseMessage with Serializable {
    val messageId = GameFightNewRoundMessage.messageId

    val roundNumber: Int

    def serialize(buf: Buffer) {
      buf.writeInt(roundNumber)
    }

  }

  protected def create(roundNumber: Int): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val roundNumber = buf.readInt()

    create(roundNumber)
  }

}

object GameFightNewRoundMessage extends GameFightNewRoundMessage {
  case class MessageImpl(roundNumber: Int) extends Message
  type Target = MessageImpl

  override def create(roundNumber: Int) = MessageImpl(roundNumber)

  def apply(roundNumber: Int) = create(roundNumber)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/GameFightNewRoundMessage.java


// Generated on 05/08/2013 19:37:44
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class GameFightNewRoundMessage extends NetworkMessage {
    public static final int MESSAGE_ID = 6239;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public int roundNumber;
    
    public GameFightNewRoundMessage() { }
    
    public GameFightNewRoundMessage(int roundNumber) {
        this.roundNumber = roundNumber;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeInt(roundNumber);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        roundNumber = buf.readInt();
        if (roundNumber < 0)
            throw new RuntimeException("Forbidden value on roundNumber = " + roundNumber + ", it doesn't respect the following condition : roundNumber < 0");
    }
    
}
